src/utils/outputs/style-dictionary/colors.ts

```diff
@@ -0,0 +1,149 @@
+import { DefaultColorType, HeisslufType, SpeakingName } from "../../data.ts";
+import { getPalette } from "../index.ts";
+import { setObjectByPath } from "./index.ts";
+
+export const getSDColorPalette = (
+  allColors: Record<string, DefaultColorType>,
+  luminanceSteps: number[],
+): any => {
+  const palette: Record<string, HeisslufType[]> = getPalette(
+    allColors,
+    luminanceSteps,
+  );
+  const colors: any = {};
+
+  Object.entries(allColors).forEach(([unformattedName, color]) => {
+    const name = unformattedName.toLowerCase();
+
+    const colorValues: any = {};
+
+    const hslType: HeisslufType[] = palette[unformattedName];
+    hslType.forEach((hsl) => {
+      colorValues[`${hsl.index ?? hsl.name}`] = { value: hsl.hex };
+    });
+
+    colorValues.origin = {
+      base: {
+        value: color.origin,
+      },
+      light: {
+        default: {
+          value: color.originLight,
+        },
+        hovered: {
+          value: color.originLightHovered,
+        },
+        pressed: {
+          value: color.originLightPressed,
+        },
+      },
+      dark: {
+        default: {
+          value: color.originDark,
+        },
+        hovered: {
+          value: color.originDarkHovered,
+        },
+        pressed: {
+          value: color.originDarkPressed,
+        },
+      },
+    };
+
+    colorValues.onOrigin = {
+      light: {
+        default: {
+          value: color.onOriginLight,
+        },
+        hovered: {
+          value: color.onOriginLightHovered,
+        },
+        pressed: {
+          value: color.onOriginLightPressed,
+        },
+      },
+      dark: {
+        default: {
+          value: color.onOriginDark,
+        },
+        hovered: {
+          value: color.onOriginDarkHovered,
+        },
+        pressed: {
+          value: color.onOriginDarkPressed,
+        },
+      },
+    };
+
+    colors[name] = colorValues;
+  });
+
+  return { colors };
+};
+
+export const getSDSpeakingColors = (
+  speakingNames: SpeakingName[],
+  allColors: Record<string, DefaultColorType>,
+): any => {
+  const colors: any = {};
+  const colorTheme = ["light", "dark"];
+  for (const [unformattedName] of Object.entries(allColors)) {
+    const name = unformattedName.toLowerCase();
+
+    colors[name] = {};
+
+    for (const theme of colorTheme) {
+      const isDark = theme === "dark";
+      const themeObj: any = {
+        origin: {
+          default: {
+            value: `colors.origin.${theme}.default.value`,
+          },
```

Comment:
**issue**: This template string resolves to `"colors.origin.light.default.value"` which is not a valid [style dictionary reference](https://amzn.github.io/style-dictionary/#/formats?id=references-in-output-files).

Wrapping them in curly braces should do the trick:

```suggestion
        origin: {
          default: {
            value: `{colors.origin.${theme}.default.value}`,
          },
```

Reply:
**issue**: I missed an error in my suggestion. References do not include the `.value` part at the end.

```tsx
`{colors.origin.${theme}.default}`,
```